packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -0,0 +1,459 @@
+import asg = require('@aws-cdk/aws-autoscaling');
+import ec2 = require('@aws-cdk/aws-ec2');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation } from './eks.generated';
+import { maxPods, nodeAmi, NodeType } from './instance-data';
+
+/**
+ * Reference properties used when a cluster is exported or imported
+ * with the const a = ClusterRef.import | export methodes
+ */
+export interface IClusterRefProps {
+  /**
+   * The physical name of the Cluster
+   */
+  clusterName: string;
+  /**
+   * The unique ARN assigned to the service by AWS
+   * in the form of arn:aws:eks:
+   */
+  clusterArn: string;
+  /**
+   * The API Server endpoint URL
+   */
+  clusterEndpoint: string;
+  /**
+   * Reeference the vpc placement for placing nodes into ASG subnets
+   */
+  vpcPlacement: ec2.VpcPlacementStrategy;
+  /**
+   * The security group ID used by the cluster for it's rules
+   */
+  securityGroupId: string;
+  /**
+   * The IConnectable interface implementation for updating
+   * security group rules
+   */
+  connections: ec2.Connections;
+}
+
+/**
+ * A SecurityGroup Reference, object not created with this template.
+ */
+export abstract class ClusterRef extends cdk.Construct implements ec2.IConnectable {
+  /**
+   * Import an existing cluster
+   *
+   * @param parent the construct parent, in most cases 'this'
+   * @param id the id or name to import as
+   * @param props the cluster properties to use for importing information
+   */
+  public static import(parent: cdk.Construct, id: string, props: IClusterRefProps): ClusterRef {
+    return new ImportedCluster(parent, id, props);
+  }
+
+  public abstract readonly clusterName: string;
+  public abstract readonly clusterArn: string;
+  public abstract readonly clusterEndpoint: string;
+  public abstract readonly vpcPlacement: ec2.VpcPlacementStrategy;
+  public abstract readonly securityGroupId: string;
+  public abstract readonly connections: ec2.Connections;
+
+  /**
+   * Export cluster references to use in other stacks
+   */
+  public export(): IClusterRefProps {
+    return {
+      clusterName: this.makeOutput('ClusterName', this.clusterName),
+      clusterArn: this.makeOutput('ClusterArn', this.clusterArn),
+      clusterEndpoint: this.makeOutput('ClusterEndpoint', this.clusterEndpoint),
+      vpcPlacement: this.vpcPlacement,
+      securityGroupId: this.securityGroupId,
+      connections: this.connections,
+    };
+  }
+
+  private makeOutput(name: string, value: any): string {
+    return new cdk.Output(this, name, { value }).makeImportValue().toString();
+  }
+}
+
+/**
+ * Properties to instantiate the Cluster
+ */
+export interface IClusterProps extends cdk.StackProps {
+  /**
+   * The VPC in which to create the Cluster
+   */
+  vpc: ec2.VpcNetworkRef;
+  /**
+   * Where to place the cluster within the VPC
+   * Which SubnetType this placement falls in
+   * @default If not supplied, defaults to public
```

Comment:
The reason I chose public (with least privilege access) was for the cost, as private subnets need the nat for cluster communication, updates and such. But it's not a difficult change, should be flexible.

For the second piece, I wanted to separate control and data plane completely. Since it's possible to keep a "Cluster" up but remove compute capacity at will. Or maybe spin up different instance types, sizes, and families. Maybe spin up a `T2 dev worker cluster` and a separate `production` but all tied to the same control plane. I'll take a look at the link, if that's still possible I will rethink it.

Thanks